### PR TITLE
Better Slack documentation

### PR DESCRIPTION
**Pull request by joonalillfors**
Documentation for how to set up Slack app

## Review Checklist

- [x] All changes are described in the PR description
- [x] New feature and breaking changes have documentation
- [x] New endpoints have API tests
- [x] New frontend features have Enzyme tests
- [x] No unnecessary or debugging code
- [x] Questionable code is clearly marked with comments
- [x] Manually tested with latest Firefox, Safari and Chrome

### CODE REVIEW

File: docs/slack.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+here.
+
+## Configuration
+
+Slack app is now created and following steps are needed so that it can
+work with Feedbacker Forum.
+
+### Creating webhook
+
+Webhook is used so that Feedbacker Forum can post messages to some Slack
```
Comment (HenrikHelenius) [inline docs/slack.md]: article

**Review by HenrikHelenius — CHANGES_REQUESTED** [review]

File: docs/slack.md (hunk 1 above)
Comment (joonalillfors) [reply docs/slack.md]: fixd

Comment (joonalillfors) [reply docs/slack.md]: Out of date

File: docs/slack.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+Slack app is now created and following steps are needed so that it can
+work with Feedbacker Forum.
+
+### Creating webhook
+
+A webhook is used so that Feedbacker Forum can post messages to some Slack
+channel via Slack app. Go to `Incoming Webhooks`, press toggle slide from
+`off` to `on` and click `Add New Webhook to Workspace`. This redirects to
+the Slack workspace and asks to which channel is Slack app allowed to post
```
Comment (HenrikHelenius) [inline docs/slack.md]: is >>, and authorize

**Review by HenrikHelenius — CHANGES_REQUESTED** [review]

File: docs/slack.md (hunk 2 above)
Comment (HenrikHelenius) [reply docs/slack.md]: also, the webhook  is created

File: docs/slack.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+and authorization. After clicking `Authorization` Webhook URL is created.
+
+### Setting up configs
+
+Running either locally or on cloud provider, webhook, clientId and clientSecret
+needs to be specified for Feedbacker Forum on some `json` config file used.
```
Comment (HenrikHelenius) [inline docs/slack.md]: on some used config file

File: docs/slack.md (hunk 2 above)
Comment (joonalillfors) [reply docs/slack.md]: fixd

File: docs/slack.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+obtained from `Basic Information` of Slack app.
+
+### Adding bot user
+
+At section `Bot Users` new bot user can be added to Slack app. Display name and
+Default username can be specified and whether the bot is always shown online or not. Clicking `Add Bot User` creates the bot.
+
+### Slash commands
+
+Slash commands can be created at `Slash Commands`-section. All available api-endpoints for slash commands can be seen on `api.md` and the path is of type
```
Comment (HenrikHelenius) [inline docs/slack.md]: add link to said doc

File: docs/slack.md (hunk 3 above)
Comment (joonalillfors) [reply docs/slack.md]: fixd

File: docs/slack.md (hunk 4 above)
Comment (joonalillfors) [reply docs/slack.md]: i will not

File: docs/slack.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+`/api/slack/commands/SOME_COMMAND`. On Slack app's `Slash Commands`-section corresponding slash commands can be created by `Create New Command`. Command, request URL to the slash command endpoint (like specified above) and short description of slash command needs to be specified here.
+
+### OAuth and permissions
+
+On section `OAuth & Permissions` needs to be created some `Redirect URLs`and `Scopes`. Click on `Add New Redirect URL` to add redirect url for Slack authentication on Feedbacker Forum. This should be `https://DOMAIN_HERE/api/slack/oauth`. Next add permission by scope from dropdown menu and select `Confirm user's identity       identity.basic`-option. Click `Save Changes`.
```
Comment (HenrikHelenius) [inline docs/slack.md]: Flip the sentence, on -> in

File: docs/slack.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+## Installing Slack app
+
+On section `Install App` click `Install App to Workspace`-button and authorize Slack app's installment to specified Slack workspace. Now Slack is ready to be used with Feedbacker Forum.
```
Comment (HenrikHelenius) [inline docs/slack.md]: in section

File: docs/slack.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+### Slash commands
+
+Slash commands can be created at `Slash Commands`-section. All available api-endpoints for slash commands can be seen on `api.md` and the path is of type
+`/api/slack/commands/SOME_COMMAND`. On Slack app's `Slash Commands`-section corresponding slash commands can be created by `Create New Command`. Command, request URL to the slash command endpoint (like specified above) and short description of slash command needs to be specified here.
+
+### OAuth and permissions
+
+On section `OAuth & Permissions` needs to be created some `Redirect URLs`and `Scopes`. Click on `Add New Redirect URL` to add redirect url for Slack authentication on Feedbacker Forum. This should be `https://DOMAIN_HERE/api/slack/oauth`. Next add permission by scope from dropdown menu and select `Confirm user's identity       identity.basic`-option. Click `Save Changes`.
+
+## Installing Slack app
```
Comment (HenrikHelenius) [inline docs/slack.md:49]: add section: sending the first slack message through the bot

**Review by HenrikHelenius — CHANGES_REQUESTED** [review]

File: docs/slack.md (hunk 5 above)
Comment (joonalillfors) [reply docs/slack.md]: fixd

File: docs/slack.md (hunk 6 above)
Comment (joonalillfors) [reply docs/slack.md]: fixd

File: docs/slack.md (hunk 7 above)
Comment (joonalillfors) [reply docs/slack.md:49]: done and donezo

Comment (joonalillfors) [reply docs/slack.md:49]: not taking any more requests for changes

File: docs/slack.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+obtained from `Basic Information` of Slack app.
+
+### Adding bot user
+
+At section `Bot Users` new bot user can be added to Slack app. Display name and
+Default username can be specified and whether the bot is always shown online or not. Clicking `Add Bot User` creates the bot.
+
+### Slash commands
+
+Slash commands can be created at `Slash Commands`-section. All available api-endpoints for slash commands can be seen on [docs/api.md](docs/api.md) and the path is of type
```
Comment (HenrikHelenius) [inline docs/slack.md]: wrong path, now writes docs/docs/api.md

**Review by HenrikHelenius — CHANGES_REQUESTED** [review]

File: docs/slack.md (hunk 8 above)
Comment (joonalillfors) [reply docs/slack.md]: fixd

**Review by HenrikHelenius — APPROVED** [review]

**Pull request merged**